test/e2e/example/example.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2019 The Submariner Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package example
+
+import (
+    "fmt"
+    "time"
+
+    "k8s.io/apimachinery/pkg/api/errors"
+    "k8s.io/client-go/kubernetes"
+
+    "github.com/rancher/submariner/test/e2e/framework"
+
+    v1 "k8s.io/api/core/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    "k8s.io/apimachinery/pkg/util/wait"
+
+    . "github.com/onsi/ginkgo"
+    . "github.com/onsi/gomega"
+)
+
+var _ = Describe("[example] Basic example to demonstrate how to write tests using the framework", func() {
+    f := framework.NewDefaultFramework("basic-example")
+    It("Should be able to list existing nodes on the cluster", func() {
+        testListingNodes(f)
+    })
+    It("Should be able to create a pod using the provided client", func() {
+        testCreatingAPod(f)
+    })
+})
+
+func testListingNodes(f *framework.Framework) {
+    for _, cs := range f.ClusterClients {
+        testListingNodesFromCluster(cs)
+    }
+}
+
+func testListingNodesFromCluster(cs *kubernetes.Clientset) {
+    nc := cs.CoreV1().Nodes()
+    By("Requesting node list from API")
+    nodes, err := nc.List(metav1.ListOptions{})
+    Expect(err).NotTo(HaveOccurred())
```

Comment:
If err occurred then we technically shouldn't access 'nodes' so should return:

     if !Expect(err).NotTo(HaveOccurred()) {
         return
     }

Reply:
No, that will stop any test execution at that point, and call the ginkgo failure handler.

I've seen that in action, and works.

Reply:
ok - I thought that didn't stop the test.